### PR TITLE
Add molecule tests, Jenkinsfile

### PR DESCRIPTION
**Pull request by nre-ableton**
- Add Python requirements files
- Add ansible-lint configuration file
- Use cache_dir from ccache_config_options
- Add molecule tests
- Set and verify group for created files/dirs
- Add Jenkinsfile
- Build pushes and pull requests on main branch
- Version 0.0.2

### CODE REVIEW

**Comment by ala-ableton** [comment]
@nre-ableton I'm interested in reviewing this PR, thanks!

File: tasks/configure.yml [hunk 1]
```diff
@@ -2,10 +2,11 @@
 - name: Ensure ccache cache directory exists
   become: true
   file:
-    path: "{{ ccache_cache_dir }}"
```
Comment (mst-ableton) [inline tasks/configure.yml:5]: Does this mean we can remove `ccache_cache_dir`?

**Review by mst-ableton — COMMENTED** [review]
Just a question.

File: tasks/configure.yml (hunk 1 above)
Comment (nre-ableton) [reply tasks/configure.yml:5]: Do you mean from our other projects which use this role? If so, yes. If you mean from this role itself, then b926b35 effectively removes this var.

Comment (mst-ableton) [reply tasks/configure.yml:5]: Got it, so previously this var had to be set by the roles which depend on it, but now it's automatically inferred.

Comment (ala-ableton) [reply tasks/configure.yml:5]: `ccache_cache_dir` was not documented as being required (on the `README.md` or elsewhere), which means that this commit is technically a bugfix. Please reword the commit message to make it clear.

File: tasks/configure.yml [hunk 2]
```diff
@@ -2,10 +2,11 @@
...
+    path: "{{ ccache_config_options['cache_dir'] }}"
     state: directory
     owner: "{{ ccache_user }}"
     mode: "0755"
+  when: ccache_config_options['cache_dir'] is defined
```
Comment (ala-ableton) [inline tasks/configure.yml:10]: Why do we have that "Ensure ccache cache directory exists" task in the first place? If that directory doesn't exist, ccache should be able to create it itself.

File: molecule/default/Dockerfile.j2 [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+{% endif %}
+
+RUN apt-get update && \
+    apt-get install -y ca-certificates build-essential gcc gnupg python3 python3-apt \
+        sudo && \
+    apt-get clean
+
+RUN useradd -G sudo -m molecule
+# Allow the molecule user to sudo without a password
+RUN echo "molecule ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
```
Comment (ala-ableton) [inline molecule/default/Dockerfile.j2:14]: Missing newline at EOF

**Review by ala-ableton — CHANGES_REQUESTED** [review]

File: molecule/default/Dockerfile.j2 (hunk 3 above)
Comment (nre-ableton) [reply molecule/default/Dockerfile.j2]: Fixed in 704fcea.

File: tasks/configure.yml (hunk 2 above)
Comment (nre-ableton) [reply tasks/configure.yml]: Ok, let's remove the task entirely. Fixed in 308c619.

File: molecule/default/tests/test_default.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import os
+import pytest
+
+import testinfra.utils.ansible_runner
```
Comment (ala-ableton) [inline molecule/default/tests/test_default.py:4]:
```suggestion

import pytest
import testinfra.utils.ansible_runner
```

**Review by ala-ableton — COMMENTED** [review]
The fixup commits look good. Please rebase/squash and reword the commit related to `ccache_cache_dir`.

File: tasks/configure.yml [hunk 5]
```diff
@@ -1,12 +1,4 @@
 ---
-- name: Ensure ccache cache directory exists
```
Comment (ala-ableton) [inline tasks/configure.yml:2]: The commit message body of 58cbfd4780d448d8454a2ecc6d56fd5e9e0bb691 mentions "duplicating data", but this commit is about removing a task that might not work (if `ccache_cache_dir` is not defined) and is not required for ccache to work.

**Review by ala-ableton — COMMENTED** [review]
Please reword 58cbfd4780d448d8454a2ecc6d56fd5e9e0bb691 to match the commit content and intent.

**Review by ala-ableton — APPROVED** [review]

**Pull request merged**